GpsApp/Top/Topology.cpp

```diff
@@ -199,37 +241,52 @@ void constructApp(int port_number, char* hostname, char* device) {
         Fw::Logger::logMsg("[INFO] Opened GPS UART: %s\n", reinterpret_cast<POINTER_CAST>(device));
         uart_connected = true;
     }
+
     // register ping table
     health.setPingEntries(pingEntries,FW_NUM_ARRAY_ELEMENTS(pingEntries),0x123);
 
     // Active component startup
     // start rate groups
-    rateGroup1Comp.start(ACTIVE_COMP_1HZ_RG, 120,10 * 1024);
-    rateGroup2Comp.start(ACTIVE_COMP_P5HZ_RG, 119,10 * 1024);
-    rateGroup3Comp.start(ACTIVE_COMP_P25HZ_RG, 118,10 * 1024);
+    rateGroup1Comp.start(0, 120,10 * 1024);
+    rateGroup2Comp.start(0, 119,10 * 1024);
+    rateGroup3Comp.start(0, 118,10 * 1024);
+    // start driver
+    blockDrv.start(0,140,10*1024);
     // start dispatcher
-    cmdDisp.start(ACTIVE_COMP_CMD_DISP,101,10*1024);
+    cmdDisp.start(0,101,10*1024);
     // start sequencer
-    cmdSeq.start(ACTIVE_COMP_CMD_SEQ,100,10*1024);
+    cmdSeq.start(0,100,10*1024);
     // start telemetry
-    eventLogger.start(ACTIVE_COMP_LOGGER,98,10*1024);
-    chanTlm.start(ACTIVE_COMP_TLM,97,10*1024);
-    prmDb.start(ACTIVE_COMP_PRMDB,96,10*1024);
+    eventLogger.start(0,98,10*1024);
+    chanTlm.start(0,97,10*1024);
+    prmDb.start(0,96,10*1024);
+
     //GPS-- GPS thread starting. The GPS component is active, so its governing thread must be started
     //      with the unique id, defined above, a priority 256 (highest) - 0 (lowest) set here to 99, and
     //      a stack size for the thread, here 10KB is used.
     gpsImpl.start(ACTIVE_COMP_GPS, 95, 10*1024);
     if (uart_connected) {
         gpsSerial.startReadThread(94, 20 * 1024);
     }
-    fileDownlink.start(ACTIVE_COMP_FILE_DOWNLINK, 100, 10*1024);
-    fileUplink.start(ACTIVE_COMP_FILE_UPLINK, 100, 10*1024);
+
+    fileDownlink.start(0, 100, 10*1024);
+    fileUplink.start(0, 100, 10*1024);
+    fileManager.start(0, 100, 10*1024);
+
+    //pingRcvr.start(0, 100, 10*1024);
+
+   
 
     // Initialize socket server if and only if there is a valid specification
     if (hostname != NULL && port_number != 0) {
-        socketIpDriver.startSocketTask(100, 10 * 1024, hostname, port_number);
+        Fw::EightyCharString name("ReceiveTask");
+        // Uplink is configured for receive so a socket task is started
+        comm.configure(hostname, port_number);
+        comm.startSocketTask(name, 100, 10 * 1024);
     }
+    return false;
 }
+/*
```

Comment:
I think you can safely remove this code - it's been replaced by the code you added to main.cpp

Does the tutorial reflect these changes?